### ISSUE TITLE
Conditions on datagram forwarding

### ISSUE THREAD

**Issue by martinthomson** [ietf-wg-masque/draft-ietf-masque-connect-udp#69]
> An intermediary that understands the request semantics enough to know that capsules are in use MAY convert between QUIC DATAGRAM frames and DATAGRAM capsules

I think that this is right, but we should be more precise about what this means.

That is, if the method is CONNECT and either:

1. the intermediary understands the :protocol selection, or
2. the request includes Capsule-Protocol

Then the intermediary can forward DATAGRAM capsules as QUIC DATAGRAM frames or forward QUIC DATAGRAM frames as DATAGRAM capsules.

Those conditions could be spelled out more clearly.

**Comment by DavidSchinazi** [comment]
I agree. I'll add this to the [PR](https://github.com/ietf-wg-masque/draft-ietf-masque-h3-datagram/pull/124).

Note that this was filed on the wrong repository, [draft-ietf-masque-h3-datagram](https://github.com/ietf-wg-masque/draft-ietf-masque-h3-datagram/issues) is the droid you're looking for.

**Comment by martinthomson** [comment]
Oops. You should be able to move the issue.
